tests/system/providers/amazon/aws/example_kinesis_analytics.py

```diff
@@ -0,0 +1,271 @@
+        stop_application,
+        await_stop_application,
+        delete_application(application_name),
+    )
+
+
+@task_group
+def copy_jar_to_s3(bucket: str):
+    """
+
+    Copy application code to S3 using HttpToS3Operator.
+
+    :param bucket: Name of the Amazon S3 bucket.
+    """
+
+    @task
+    def create_connection(conn_id):
+        conn = Connection(
+            conn_id=conn_id,
+            conn_type="http",
+            host="https://github.com/",
+        )
+        session = settings.Session()
+        session.add(conn)
+        session.commit()
+
+    @task(trigger_rule=TriggerRule.ALL_DONE)
+    def delete_connection(conn_id: str):
+        session = settings.Session()
+        conn_to_details = session.query(Connection).filter(Connection.conn_id == conn_id).first()
+        session.delete(conn_to_details)
+        session.commit()
+
+    copy_jar_file = HttpToS3Operator(
+        task_id="copy_jar_file",
+        http_conn_id=http_conn_id,
+        endpoint="awslabs/managed-service-for-apache-flink-blueprints/releases/download/v2.0.1/kds-to-s3-datastream-java-1.0.1.jar",
+        s3_bucket=bucket,
+        s3_key="code/kds-to-s3-datastream-java-1.0.1.jar",
+    )
+
+    chain(create_connection(http_conn_id), copy_jar_file, delete_connection(http_conn_id))
+
+
+@task
+def create_kinesis_stream(stream: str):
+    """
+    Create kinesis stream and put some sample data.
+
+    :param stream: Name of the kinesis stream.
+    """
+    client = boto3.client("kinesis", region_name=region_name)
+    client.create_stream(StreamName=stream, ShardCount=1, StreamModeDetails={"StreamMode": "PROVISIONED"})
+    account_id = boto3.client("sts").get_caller_identity()["Account"]
+    waiter = client.get_waiter("stream_exists")
+    waiter.wait(StreamName=stream, WaiterConfig={"Delay": 60, "MaxAttempts": 4})
+
+    def get_data():
+        return {
+            "event_time": dt.datetime.now().isoformat(),
+            "ticker": random.choice(["AAPL", "AMZN", "MSFT", "INTC", "TBV"]),
+            "price": round(random.random() * 100, 2),
+        }
+
+    for _ in range(2):
+        data = get_data()
+        client.put_record(
+            StreamARN=f"arn:aws:kinesis:{region_name}:{account_id}:stream/{stream}",
+            Data=json.dumps(data),
+            PartitionKey=data["ticker"],
+        )
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def delete_kinesis_stream(stream: str):
+    client = boto3.client("kinesis", region_name=region_name)
+    client.delete_stream(StreamName=stream, EnforceConsumerDeletion=True)
+
+
+with DAG(
+    dag_id=DAG_ID,
+    schedule="@once",
+    start_date=datetime(2021, 1, 1),
+    tags=["example"],
+    catchup=False,
+) as dag:
+    test_context = sys_test_context_task()
+    env_id = test_context["ENV_ID"]
+    bucket_name = f"{env_id}-kinesis-analytics"
+    application_name = f"{env_id}-test-app"
+    http_conn_id = f"{env_id}-git"
+    region_name = boto3.session.Session().region_name
```

Comment:
`region_name` is `magically` globally available to all the functions defined above. Maybe passing it as parameter would make it less confusing?

Reply:
Updated thank you 😄